### ISSUE TITLE
[Bug]: Incompatibilty issue with file_picker

### ISSUE THREAD

**Issue by kavania2002** [fluttercommunity/plus_plugins#1889]
### Platform

Android 12

### Plugin

package_info_plus

### Version

3.0.3

### Flutter SDK

3.10.2

### Steps to reproduce

1. Add following dependencies into the pubspec.yaml file:

```
  cupertino_icons: ^1.0.2
  percent_indicator: ^4.2.2
  datetime_picker_formfield: ^2.0.0
  video_player: ^2.4.4
  simple_horizontal_stepper: ^0.0.1
  url_launcher: ^6.0.17
  provider: ^6.0.3
  firebase_auth: ^4.2.2
  country_picker: ^2.0.15
  sms_autofill: ^2.2.0
  pinput: ^2.2.11
  http: ^0.13.4
  email_validator: ^2.0.1
  shared_preferences: ^2.0.15
  camera: ^0.10.0
  image_picker: ^0.8.4+11
  file_picker: ^4.5.1
  open_file_safe: ^3.2.1
  path_provider: ^2.0.9
  dio: ^4.0.6
  google_mlkit_pose_detection: ^0.5.0
  vector_math: any
  mime: ^1.0.2
  intl_phone_field: ^3.1.0
  chewie: ^1.3.6
  auto_orientation: ^2.2.1
  flick_video_player: ^0.5.0
  mockito: ^5.3.2
  visibility_detector: ^0.3.3
  firebase_messaging: ^14.2.0
  device_info: ^2.0.3
  permission_handler: ^10.2.0
  intl: ^0.17.0
  validators: ^3.0.0
  date_time_picker: ^2.0.0
  package_info_plus: ^3.0.3
```

2. ```flutter pub get```

**NOTE**: Initially the problem was same as mentioned in #1525. So to avoid the error, I upgraded to version ```^3.0.3``` and ran into dependencies error. 

### Code Sample

_No response_

### Logs

```shell
Here's the log:


fact: win32 >=2.7.0 <3.0.0-∞ depends on ffi ^2.0.0
SLVR:                                                                                                                                             
                                                                                                                                                  
  selecting win32 2.7.0
SLVR:                                                                                                                                             
                                                                                                                                                  
    selecting wakelock_web 0.4.0
SLVR:                                                                                                                                             
                                                                                                                                                  
      selecting wakelock_platform_interface 0.3.0
SLVR:                                                                                                                                             
                                                                                                                                                  
        selecting wakelock_macos 0.4.0
SLVR:                                                                                                                                             
                                                                                                                                                  
          selecting platform 3.1.0
SLVR:                                                                                                                                             
                                                                                                                                                  
            selecting xdg_directories 1.0.0
SLVR:                                                                                                                                             
                                                                                                                                                  
            derived: process ^4.0.0
SLVR:                                                                                                                                             
                                                                                                                                                  
              selecting petitparser 5.4.0
SLVR:                                                                                                                                             
                                                                                                                                                  
                selecting pointycastle 3.7.3
SLVR:                                                                                                                                             
                                                                                                                                                  
                  selecting fixnum 1.1.0
SLVR:                                                                                                                                             
                                                                                                                                                  
                    selecting process 4.2.4
SLVR:                                                                                                                                             
                                                                                                                                                  
                    fact: file_picker >=4.0.0 <5.0.0-∞ depends on ffi ^1.1.2
SLVR:                                                                                                                                             
                                                                                                                                                  
                    fact: file_picker >=3.0.0-nullsafety.0 depends on flutter from sdk
SLVR:                                                                                                                                             
                                                                                                                                                  
                    fact: file_picker >=4.0.1 <5.0.0-∞ depends on flutter_plugin_android_lifecycle ^2.0.2
SLVR:                                                                                                                                             
                                                                                                                                                  
                    fact: file_picker >=3.0.0-nullsafety.0 depends on flutter_web_plugins from sdk
SLVR:                                                                                                                                             
                                                                                                                                                  
                    fact: file_picker >=4.0.0 <5.2.11-∞ depends on path ^1.8.0
SLVR:                                                                                                                                             
                                                                                                                                                  
                    fact: file_picker >=3.0.0-nullsafety.3 <5.0.0-∞ depends on plugin_platform_interface ^2.0.0
SLVR:                                                                                                                                             
                                                                                                                                                  
                    fact: file_picker >=4.5.0 <5.0.0-∞ depends on win32 ^2.4.1
SLVR:                                                                                                                                             
                                                                                                                                                  
                    conflict: file_picker >=4.0.0 <5.0.0-∞ depends on ffi ^1.1.2
SLVR:                                                                                                                                             
                                                                                                                                                  
                    ! not ffi ^1.1.2 is satisfied by ffi ^2.0.1
SLVR:                                                                                                                                             
                                                                                                                                                  
                    ! which is caused by "package_info_plus >=3.0.0 depends on ffi ^2.0.1"
SLVR:                                                                                                                                             
                                                                                                                                                  
                    ! thus: file_picker >=4.0.0 <5.0.0-∞ is incompatible with package_info_plus >=3.0.0
SLVR:                                                                                                                                             
                                                                                                                                                  
                    ! file_picker >=4.0.0 <5.0.0-∞ is satisfied by file_picker ^4.5.1
SLVR:                                                                                                                                             
                                                                                                                                                  
                    ! which is caused by "urhab depends on file_picker ^4.5.1"
SLVR:                                                                                                                                             
                                                                                                                                                  
                    ! thus: package_info_plus >=3.0.0 is forbidden
SLVR:                                                                                                                                             
                                                                                                                                                  
                    ! package_info_plus >=3.0.0 is satisfied by package_info_plus ^3.0.3
SLVR:                                                                                                                                             
                                                                                                                                                  
                    ! which is caused by "urhab depends on package_info_plus ^3.0.3"
SLVR:                                                                                                                                             
                                                                                                                                                  
                    ! thus: version solving failed
SLVR: Version solving took 0:00:05.644906 seconds.
    | Tried 3 solutions.
FINE: Resolving dependencies finished (5.766s).
ERR : Because file_picker ^4.0.0 depends on ffi ^1.1.2 and package_info_plus >=3.0.0 depends on ffi ^2.0.1, file_picker ^4.0.0 is incompatible with
    |   package_info_plus >=3.0.0.
```
```


### Flutter Doctor

```shell
[✓] Flutter (Channel stable, 3.10.2, on Microsoft Windows [Version 10.0.22621.1778], locale en-IN)
    • Flutter version 3.10.2 on channel stable at C:\dev\flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 9cd3d0d9ff (9 days ago), 2023-05-23 20:57:28 -0700
    • Engine revision 90fa3ae28f
    • Dart version 3.0.2
    • DevTools version 2.23.1

[✓] Windows Version (Installed version of Windows is version 10 or higher)

[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.2)
    • Android SDK at C:\Users\kavan\AppData\Local\Android\sdk
    • Platform android-33, build-tools 33.0.2
    • Java binary at: C:\Program Files\Android\Android Studio\jbr\bin\java
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-b2043.56-9586694)
    • All Android licenses accepted.

[✗] Chrome - develop for the web (Cannot find Chrome executable at .\Google\Chrome\Application\chrome.exe)
    ! Cannot find Chrome. Try setting CHROME_EXECUTABLE to a Chrome executable.

[✗] Visual Studio - develop for Windows
    ✗ Visual Studio not installed; this is necessary for Windows development.
      Download at https://visualstudio.microsoft.com/downloads/.
      Please install the "Desktop development with C++" workload, including all of its default components

[✓] Android Studio (version 2022.2)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-b2043.56-9586694)

[✓] VS Code (version 1.78.2)
    • VS Code at C:\Users\kavan\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.66.0

[✓] Connected device (2 available)
    • Windows (desktop) • windows • windows-x64    • Microsoft Windows [Version 10.0.22621.1778]
    • Edge (web)        • edge    • web-javascript • Microsoft Edge 113.0.1774.57

[✓] Network resources
    • All expected network resources are available.

! Doctor found issues in 2 categories.
```


### Checklist before submitting a bug

- [X] I Google'd a solution and I couldn't find it
- [X] I searched on StackOverflow for a solution and I couldn't find it
- [X] I read the README.md file of the plugin
- [X] I'm using the latest version of the plugin
- [X] All dependencies are up to date with `flutter pub upgrade`
- [X] I did a `flutter clean`
- [X] I tried running the example project

**Comment by kavania2002** [comment]
You were right. 
Though updating the version still gives dependency errors, but it's because of another package ```open_file_safe```

```
Because package_info_plus >=3.0.0 depends on ffi ^2.0.1 and open_file_safe 3.2.3 depends on ffi ^1.0.0, package_info_plus >=3.0.0 is incompatible with open_file_safe 3.2.3.
```

**Comment by vbuberen** [comment]
Yes, but still it is `open_file_safe` that wasn't updated last 17 months and relies on older versions of `ffi`.
I would suggest to report the issue in `open_file_safe` asking the maintainer to update the plugin.

Or you can try to use dependency_overrides and fix `ffi` version to `2.0.1`, for example. You can read more about overriding dependencies here: https://dart.dev/tools/pub/dependencies#dependency-overrides

Closing as it is not a bug on `package_info_plus` side`.
